realm/realm-library/src/androidTest/java/io/realm/RealmQueryTests.java

```diff
@@ -2982,13 +2644,10 @@ public void isEmpty_acrossLink_wrongTypeThrows() {
 
             RealmQuery<Owner> query = realm.where(Owner.class);
             try {
-                query.isEmpty(Owner.FIELD_CAT + "." + Cat.FIELD_AGE);
+                query.isEmpty(Owner.FIELD_CAT + "." + Cat.FIELD_AGE).findAll();
```

Comment:
If this doesn't throw without the `.findAll` then we have changed the semantics -> CHANGELOG